.travis.yml

```diff
@@ -1,10 +1,26 @@
 services:
   - docker
 
+branches:
+  only:
+    - master
+    - module-8
+
 script:
   - docker build --target test --tag my-test-image . 
   - docker run my-test-image tests
   - docker run --env SECRET_KEY --env TRELLO_TODOID --env TRELLO_DONEID --env TRELLO_BOARDID --env TRELLO_KEY --env TRELLO_TOKEN my-test-image test_e2e
+before_deploy:
+  - docker login --username $DOCKER_USERNAME --password $DOCKER_PASSWORD
+  - docker build --target production --tag aj1993/my-image-prod:latest .
+  - docker push aj1993/my-image-prod:latest
+  - docker login --username=_ --password=$HEROKU_API_TOKEN registry.heroku.com
+deploy:
+  provider: script
+  script: docker pull aj1993/my-image-prod:latest &&  docker tag aj1993/my-image-prod:latest registry.heroku.com/aptem-trello-app/web && docker push registry.heroku.com/aptem-trello-app/web && heroku cotainer:release --app aptem-trello-app
```

Comment:
> `poetry install` throws an error for me, looks like maybe you haven't committed changes to `poetry.lock` since installing gunicorn? If you don't have the changes any more try running `poetry update` to fix it.
> Your CI pipeline still works because you don't copy `poetry.lock` into your Docker image before doing `poetry install` - you should or you could end up with different versions of your dependencies on your "production" app and find it breaks when it works locally.

Hi Ben, Don't think I made any poetry.lock changes

Reply:
Poetry will automatically update the `poetry.lock` file when you add or update a dependency and you should commit those changes.

Check you don't have any uncommitted changes (`git status`), then try running `poetry install` and see what happens - I expect you'll get a warning about the lock file being out of date.

Reply:
I checked git status. Said everything was up to date

Reply:
Doesn't work. So I should copy it in what line of the dockerfile?

Reply:
Actually I wonder if I left it out deliberatley

Reply:
Run `poetry update` and then commit the changes it makes to `poetry.lock`. That should bring it back in sync and fix `poetry install`.

In terms of the Dockerfile, you'd want to COPY it in just before you `RUN poetry install`, so the same time you copy `pyproject.toml`.

Reply:
So under line 6? Got it

Reply:
I think it should be okay now. It did pass Travis.

Reply:
No wait it's still building

Reply:
It works now.

Reply:
Wait. No. The final line of the Travis Deploy is

`Skipping a deployment with the script provider because this branch is not permitted: module-8`

Reply:
So I don't know if this has worked or not

Reply:
Line 17 looks like it should work if `$HEROKU_API_TOKEN` is valid.
Double check `$HEROKU_API_TOKEN` is set correctly in your Travis settings. You could also try changing the line to
```yaml
  - heroku container:login
```
which will use `$HEROKU_API_KEY` instead.

Reply:
Thanks. It works as a deploy now.